L1Trigger/DTTriggerPhase2/interface/CandidateGroup.h

```diff
@@ -0,0 +1,65 @@
+#ifndef L1Trigger_DTTriggerPhase2_CandidateGroup_h
+#define L1Trigger_DTTriggerPhase2_CandidateGroup_h
+
+#include <tuple>
+#include <vector>
+#include <bitset>
+#include <iostream>
+//Note that this being in src is probably wrong but I don't want to crash everone's code for being picky
+#include "L1Trigger/DTTriggerPhase2/interface/dtprimitive.h"
+#include "L1Trigger/DTTriggerPhase2/interface/Pattern.h"
+
+class CandidateGroup {
+public:
+  //Constructors and destructors
+  CandidateGroup();
+  CandidateGroup(Pattern* p);
+  virtual ~CandidateGroup();
+
+  //Hit operation procedures
+  void addHit(DTPrimitive dthit, int lay, bool isGood);
+  void removeHit(DTPrimitive dthit);
+
+  //Get Methods
+  int candId() const { return candId_; };
+  int nhits() const { return nhits_; };
+  int nisGood() const { return nisGood_; };
+  int nLayerhits() const { return nLayerhits_; };
+  int nLayerUp() const { return nLayerUp_; };
+  int nLayerDown() const { return nLayerDown_; };
+  std::vector<DTPrimitive> candHits() const { return candHits_; };
```

Comment:
Hi @makortel , I need candHits() not to be declared as const ref because I'll be adding more info to candHits_ later on on L1Trigger/DTTriggerPhase2/src/PseudoBayesGrouping.cc

Reply:
Yes, if you modify the `DTPrimitive` then it needs be a non-`const` reference, but then the method itself must be non-`const` as well. I suggested `const` reference because it is functionally equivalent to a copy for the calling code for the cases a copy is really needed, and allows avoiding the copy when copy is not needed.

Is this loop the one where you modify the `candHits_`?
https://github.com/cms-sw/cmssw/blob/92ace81d6d923fe5246e87b6e7197928cee9e677/L1Trigger/DTTriggerPhase2/src/PseudoBayesGrouping.cc#L208-L209
Note that currently the loop definition itself leads to undefined behavior (two calls to `candHits()` lead to two different vectors, and comparing iterators of two vectors is undefined behavior). Also note that with a copy, all modifications to the contained `DTPrimitive`s are lost after the loop (unless `DTPrimitive` holds  those values by pointers, in which case this method should be non-`const` as well).